packages/features/bookings/lib/handleNewBooking/test/booking-limits.test.ts

```diff
@@ -1,7 +1,464 @@
-import { describe } from "vitest";
+/**
+ * These integration tests aim to cover difficult-to-test edge cases
+ * Standard cases are currently handled in e2e tests only
+ *
+ * see: https://github.com/calcom/cal.com/pull/10480
+ *      https://github.com/calcom/cal.com/pull/10968
+ */
+import prismock from "../../../../../../tests/libs/__mocks__/prisma";
 
+import { describe, expect, vi } from "vitest";
+
+import { BookingStatus } from "@calcom/prisma/enums";
 import { test } from "@calcom/web/test/fixtures/fixtures";
+import {
+  TestData,
+  createBookingScenario,
+  getBooker,
+  getDate,
+  getNextMonthNotStartingOnWeekStart,
+  getOrganizer,
+  getScenarioData,
+} from "@calcom/web/test/utils/bookingScenario/bookingScenario";
+import { expectBookingToBeInDatabase } from "@calcom/web/test/utils/bookingScenario/expects";
+
+import { createMockNextJsRequest } from "./lib/createMockNextJsRequest";
+import { getMockRequestDataForBooking } from "./lib/getMockRequestDataForBooking";
+import { setupAndTeardown } from "./lib/setupAndTeardown";
+
+// Local test runs sometime gets too slow
+const timeout = process.env.CI ? 5000 : 20000;
+
+const eventLength = 30;
```

Comment:
Using this to set `slotInterval` and `length`, and also to calculate duration limits. Just flagging as those two eventType props were set to `45` in other integration tests.